packages/react-component-library/src/components/Alert/Alert.test.tsx

```diff
@@ -203,4 +203,20 @@ describe('Alert', () => {
       )
     })
   })
+
+  describe('when hideDismiss prop is provided', () => {
+    beforeEach(() => {
+      wrapper = render(<Alert hideDismiss>Description</Alert>)
+    })
+
+    it('should not render the close button', () => {
+      expect(wrapper.queryByTestId('close')).not.toBeInTheDocument()
```

Comment:
I think this is fine as it's consistent with other testing but I might be tempted to start using `getByRole('button', { name: 'Close' }`